### ISSUE TITLE
Time with microseconds in system.query_log

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#9800]
and other system logs.

This feature is requested by Yandex.Metrica.

**Comment by bharatnc** [comment]
I have made an initial PR in  #14252  to introduce 1. `query_start_time_microseconds ` for the `system.query_log` table. I will be doing the same to these: (hopefully in separate PRs)

1. `system.query_thread_log` -  will add a new field called `query_start_time_microseconds ` (DateTime64). 

(One question is that for `system.query_log` and `system.query_thread_log` - do we really need `event_time_microseconds` field ?)

Otherwise, for all these tables, the only relevant addition seems to be `event_time_microseconds` (`DateTime64`):

2. `system.trace_log`  
3. `system.text_log`  
4.  `system.asynchronous_metric_log`  
5. `system.metric_log` 
6. `system.part_log`

(Format will look like  `2020-08-29 21:15:47.667301`.)

@alexey-milovidov  as far as I have looked, those seem to be the system_log tables & fields that require this time in microseconds - I might be wrong.  What do you think?

_PS:_ The naming recommendation lines up with your recommendation in  #4821 which might be a duplicate of this issue.

**Comment by alexey-milovidov** [comment]
Better to add microsecond variants for both event_time and query_start_time.
It may look somewhat redundant in presence of `query_duration_ms` but it's better for consistency.

Yes, let's do this change for all the tables.

**Comment by bharatnc** [comment]
Sounds good 👍 thank you for confirming!

**Comment by filimonov** [comment]
@bharatnc @alexey-milovidov is it resolved and can be closed? Or there is smth more to do here?

**Comment by bharatnc** [comment]
@filimonov  one last table left that I am working on `system.part_log` which I have started to work on. After that this can be closed.

**Comment by bharatnc** [comment]
> @bharatnc @alexey-milovidov is it resolved and can be closed? Or there is smth more to do here?

Last PR is open. We can close this out once that is merged.

**Comment by bharatnc** [comment]
@filimonov  @alexey-milovidov I think this can be closed. All have `event_time_microseconds` and `query_time_microseconds` field except the following case:

I added another new table a month back [system.distributed_ddl_queue](https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/operations/system-tables/distributed_ddl_queue.md) for which `event_time_microseconds` column is not present as `event_time` is parsed from `mtime` from `zookeeper`: https://github.com/ClickHouse/ClickHouse/blob/master/src/Common/ZooKeeper/IKeeper.h which is UInt64 and I need a mtime with `Decimal64` for `event_time_microseconds` to work. I think that can be added if needed. It's still possible to query `event_time_microseconds` from `system.query_log` given the `entry` (query_id) from the `system.distributed_ddl_table` for a particular entry.
